### PR TITLE
Compile the project against Play 2.4.0-RC3 to fix binary incompatibil…

### PR DESCRIPTION
**Pull request by raelg**
…ities introduced in new Play version #35

### CODE REVIEW

**Comment by raelg** [comment]
This was not as simple as I initially expected. Working on fixing the tests.

File: build.sbt [hunk 1]
```diff
@@ -8,14 +7,20 @@ import xerial.sbt.Sonatype._
...
-version := "0.4-SNAPSHOT"
+version := "0.4.1"
 
 libraryDependencies ++= Seq(
   "com.googlecode.htmlcompressor" % "htmlcompressor" % "1.5.2",
   "rhino" % "js" % "1.7R2"
 )
 
-lazy val root = (project in file(".")).enablePlugins(PlayScala)
+libraryDependencies += specs2 % Test
```
Comment (akkie) [inline build.sbt]: Please could you add this dependency to the dependency list above.

File: build.sbt [hunk 2]
```diff
@@ -8,14 +7,20 @@ import xerial.sbt.Sonatype._
...
+
+libraryDependencies += "org.easytesting" % "fest-assert-core" % "2.0M10"
```
Comment (akkie) [inline build.sbt]: If this library is really needed? If yes then you should add it to the dependency list above and mark it as testing library.

File: test/com/mohiva/play/htmlcompressor/java/HTMLCompressorFilterTest.java [hunk 3]
```diff
@@ -185,10 +185,12 @@ public void run() {
...
                 Result result = route(fakeRequest(GET, "/static"));
 
-                assertThat(status(result)).isEqualTo(OK);
-                assertThat(contentType(result)).isEqualTo("text/html");
-                assertThat(contentAsString(result)).startsWith("<!DOCTYPE html><html><head>");
-                assertThat(header(CONTENT_LENGTH, result)).isNotEqualTo(String.valueOf(file.length()));
+                assertThat(result.status()).isEqualTo(OK);
+                assertThat(result.contentType()).isEqualTo("text/html");
+                String actual = contentAsString(result);
+                System.out.println("actual = " + actual);
```
Comment (akkie) [inline test/com/mohiva/play/htmlcompressor/java/HTMLCompressorFilterTest.java]: Should be removed

File: build.sbt [hunk 4]
```diff
@@ -8,14 +7,20 @@ import xerial.sbt.Sonatype._
 
 name := "play-html-compressor"
 
-version := "0.4-SNAPSHOT"
+version := "0.4.1"
```
Comment (akkie) [inline build.sbt]: Version should be a snapshot version until Play 2.4 final will be released. So the version should be 0.4.1-SNAPSHOT

**Comment by akkie** [comment]
Thanks for your work on this. Looks good to me expect the parts I've commented. Could you please squash the commits into a single commit after you have updated the changes.

File: build.sbt (hunk 2 above)
Comment (raelg) [inline build.sbt]: Yeah this was changed according to the migration guide: https://www.playframework.com/documentation/2.4.0-RC1/Migration24 
Oddly that paragraph was removed from the 2.4.x link https://www.playframework.com/documentation/2.4.x/Migration24

Comment (raelg) [inline build.sbt]: With regards to fest you were right - it was a problem with my IDE setup.

File: build.sbt [hunk 5]
```diff
@@ -8,14 +8,19 @@ import xerial.sbt.Sonatype._
...
 
-version := "0.4-SNAPSHOT"
+version := "0.4.1-SNAPSHOT"
 
 libraryDependencies ++= Seq(
   "com.googlecode.htmlcompressor" % "htmlcompressor" % "1.5.2",
-  "rhino" % "js" % "1.7R2"
+  "rhino" % "js" % "1.7R2",
+  specs2 % Test,
+  "org.easytesting" % "fest-assert" % "1.4" % Test
```
Comment (akkie) [inline build.sbt]: I thought this is not needed?

Comment (raelg) [inline build.sbt]: The dependency appears to have been removed from the default set

**Comment by akkie** [comment]
@raelg Thanks! :thumbsup:

**Pull request merged**